source/opt/optimizer.cpp

```diff
@@ -571,6 +571,25 @@ bool Optimizer::RegisterPassFromFlag(const std::string& flag,
              pass_args.c_str());
       return false;
     }
+  } else if (pass_name == "struct-packing") {
+    if (pass_args.size() == 0) {
+      Error(consumer(), nullptr, {},
+            "--struct-packing requires a name:rule argument.");
+      return false;
+    }
+
+    auto separator_pos = pass_args.find(':');
+    if (separator_pos == std::string::npos || separator_pos == 0 ||
+        separator_pos + 1 == pass_args.size()) {
+      Errorf(consumer(), nullptr, {},
+             "Invalid argument for --struct-packing: %s", pass_args.c_str());
+      return false;
+    }
+
+    const std::string struct_name = pass_args.substr(0, separator_pos);
+    const std::string rule_name = pass_args.substr(separator_pos + 1);
+
+    RegisterPass(CreateStructPackingPass(struct_name.c_str(), rule_name.c_str()));
```

Comment:
What happens if someone add an invalid rule name?

Reply:
I see a mix of Google and KhronosGroup or even other companies' name in the other passes. Should I use the same as [SPIRV-Tools' license](https://github.com/KhronosGroup/SPIRV-Tools?tab=readme-ov-file#licence) or even Epic's?

Reply:
I replied in the other comment